arkouda/join.py

```diff
@@ -137,19 +138,8 @@ def gen_ranges(starts: pdarray, ends: pdarray) -> Tuple[pdarray, pdarray]:
     ranges : pdarray, int64
         The actual ranges, flattened into a single array
     """
-    if starts.size != ends.size:
-        raise ValueError("starts and ends must be same size")
-    if starts.size == 0:
-        return zeros(0, dtype=akint64), zeros(0, dtype=akint64)
-    lengths = ends - starts
-    if not (lengths > 0).all():
-        raise ValueError("all ends must be greater than starts")
-    segs = cumsum(lengths) - lengths
-    totlen = lengths.sum()
-    slices = ones(totlen, dtype=akint64)
-    diffs = concatenate((array([starts[0]]), starts[1:] - starts[:-1] - lengths[:-1] + 1))
-    slices[segs] = diffs
-    return segs, cumsum(slices)
+    # only maintain one version of gen_ranges
+    return seg_gen_ranges(starts, ends)
```

Comment:
Looking at the configuration of the code, I am wonder if we shouldn't just delete the function from here and only maintain it in one place. Since they are named identically, there does not seem to be a need to maintain both

Reply:
I agree, I think it would be better to delete. The only reason I didn't was if users specifically import from the module i.e. `from join import gen_ranges`. I imagine this is pretty rare and they'd just do `ak.gen_ranges`

Reply:
I actually decided to just remove the version in `segarray.py` since the one in `join` is the version that `ak.gen_ranges` points to. Now in segarry we just do `from join import gen_ranges`